File: 42-sort-colors/sort.rb
# @param {Integer[]} nums
# @return {Void} Do not return anything, modify nums in-place instead.
def sort_colors(nums)
    white,red,blue = 0,0,0
        
        for num in nums
            if num ==0
                white+=1
            end
            if num==1
                red+=1
            end
        end
    
        blue = nums.size - red-white
        nums.clear()
    
        for i in (0..white-1) 
            nums.append(0)
        end
        for i in (0..red-1)
            nums.append(1)
        end
        for i in (0..blue-1)
            nums.append(2)
        end
end

File: 42-sort-colors/sort.ts
/**
 Do not return anything, modify nums in-place instead.
 */
function sortColors(nums: number[]): void {
    let [red,white,blue] = [0,0,0]

    for(let num of nums){ 
        if(num == 0){ white++}
        else if ( num == 1){ red++}
    }


    blue = (nums.length) - (white) - (red) 

    nums.length=0 

    for(let i=0;i<white;i++){ nums.push(0)} 
    for(let i=0;i<red;i++){ nums.push(1)}
    for(let i=0;i<blue;i++){ nums.push(2)  }
};